Feature/crm system users

keydmitrij opened the pull request.

keydmitrij committed:
Change profile tab icon and add users tab

keydmitrij committed:
Create api call for fetching users list

keydmitrij committed:
Fix bug in api call

keydmitrij committed:
Fix yet another bug in api

keydmitrij committed:
Create page with list of users

keydmitrij committed:
Remove unused import

keydmitrij committed:
Create date formatting utility function and use it on users dates

keydmitrij committed:
Create one common axios instance instead of redeclaring it in every api

keydmitrij committed:
Remove trash dependency that creates infinite loop

keydmitrij committed:
Change table size

keydmitrij committed:
Fix bug with logout on f5

keydmitrij committed:
Fix logout on F5

keydmitrij committed:
Add missing argument in login function

keydmitrij committed:
Change users table column order

keydmitrij committed:
Add webstorm data to gitignore

keydmitrij committed:
Add interceptor on 401 response status

Im not sure if it works :)

keydmitrij committed:
Fix bug with interceptor not throwing an error

keydmitrij committed:
Add icons for email and phone number in users list

keydmitrij committed:
Add phone formatting for users list

keydmitrij committed:
Align table columns width

Its needed to prevent overflow on long names or low resolutions

keydmitrij committed:
Add ability to collapse sidebar

keydmitrij committed:
Make todo list collapsible

keydmitrij committed:
Make sidebar collapse on huge page scaling

keydmitrij committed:
Make todo list narrower

keydmitrij committed:
Change table size

keydmitrij committed:
Create dummy layout for filter fields

keydmitrij committed:
Add icon to search field and change placeholder

keydmitrij committed:
Move getUsersList to where it belongs

keydmitrij committed:
Connect lodash debounce and add filter by name/email feature

keydmitrij committed:
Add spin loading on users list fetch in input

keydmitrij committed:
Create usersFilter state and bind it to useEffect

Я накончал в штаны когда допер до этого,
буквально общий рычаг для всех фильтров и сразу апдейтит страницу
и дает пагинацию

keydmitrij committed:
Add skeleton instead of dummy loading message

keydmitrij committed:
Add loading effect for table

keydmitrij committed:
Fix pagination page to -1 as it should be

keydmitrij committed:
Add page size filtering

keydmitrij committed:
Add sorting feature to table

keydmitrij committed:
Add correct typing

keydmitrij committed:
Fix filters persistence

Filters state now is getting construct from previous filters + new ones
it makes all filters persist on individual filters change

keydmitrij committed:
Make 401 error on interceptor retry force logout

keydmitrij committed:
Move userData to redux state

keydmitrij committed:
Move profile fetching so it doesnt prevent loading page

keydmitrij committed:
Fix typo in menu item label

keydmitrij committed:
Remove commented unused code

keydmitrij committed:
Revert "Fix typo in menu item label"

This reverts commit 7cb591c0d072ec542741fbbeee07bb3decfbdfee.

keydmitrij committed:
Add filter by block status

keydmitrij committed:
Fix pagination bug when changing isblocked filter

keydmitrij committed:
Make table columns fixed width

keydmitrij committed:
Adjust table columns width and make roles wrap

keydmitrij committed:
Add go to user button

keydmitrij committed:
Create api call for getting user by id

keydmitrij committed:
Add navigation to user profile feature

keydmitrij committed:
Change name for current user profile page

keydmitrij committed:
Create api call for updating userdata

keydmitrij committed:
Fix wrong method usage in api call

keydmitrij committed:
Create MVP of updating userdata

keydmitrij committed:
Create go back button on profile page

keydmitrij committed:
Shrink registration date column width

keydmitrij committed:
Merge branch 'feature/CRM-System-users' into feature/CRM-System-users-temp

keydmitrij committed:
Merge pull request #7 from typeshaper/feature/CRM-System-users-temp

Feature/crm system users temp

keydmitrij committed:
Fix logout bug

keydmitrij committed:
Change layout of personal info list

keydmitrij committed:
Create updating profile functional but with types error

keydmitrij committed:
Add validation rules to profile change

keydmitrij committed:
Add dummy symbol for unspecified phone number

keydmitrij committed:
Clear readme file

keydmitrij committed:
Add colors for button on profile page

keydmitrij committed:
Add typecasting to diff function

keydmitrij committed:
Change typecasting slightly

keydmitrij committed:
Add lodash and dont send request if no data is changed in user profile

keydmitrij committed:
Nothing was changed wtf git

keydmitrij committed:
Create api call for deleting user

keydmitrij committed:
Create delete user feature

keydmitrij committed:
Specify correct return type on api call

keydmitrij committed:
Add rowkey to table

keydmitrij committed:
Add modal to deleting user and get rid of disabling button after deleting

keydmitrij committed:
Create api call for blocking user

keydmitrij committed:
Fix api call method

keydmitrij committed:
Add function to block users

keydmitrij committed:
Remove trash code

keydmitrij committed:
Can block only if user is not blocked

keydmitrij committed:
Show block button for both blocked and unblocked users

keydmitrij committed:
Create api call for unlocking user

keydmitrij committed:
Add functionality to block / unblock users

keydmitrij committed:
Make action cell narrower

keydmitrij committed:
Make dropdown menu getting triggered by press not hover

keydmitrij committed:
Make dropdown menu say block if user is unlocked and unlock if blocked

keydmitrij committed:
Fix api call to unblocking user

keydmitrij committed:
Move types to another folder and add default color for roles on userspage

keydmitrij committed:
Fix import

keydmitrij committed:
Add api call for changing user roles

keydmitrij committed:
Remove unused comments

keydmitrij committed:
Create MVP of user roles selection

keydmitrij committed:
Remove unused imports

keydmitrij committed:
Add typing to useselector

keydmitrij committed:
Add right state for auth store and navigate out of users page is has no rights

keydmitrij committed:
Prohibit checking user profile if has no rights

keydmitrij committed:
Hide users page from not admin/moderator

keydmitrij committed:
Show user not found is user doesn't exist

keydmitrij committed:
Move filters logic from usersPage to redux store

keydmitrij committed:
Organize imports

keydmitrij committed:
Rename unblocked to active users and change order in status filter

keydmitrij committed:
Change + - to active and blocked for uses status

keydmitrij committed:
Add await statement to changing roles

Its absence used to skip catch block and give success message even though api call was rejected

keydmitrij committed:
Change roles to edit roles and make it so only admin can do it

keydmitrij committed:
Fix typo

keydmitrij committed:
Swap blocked and active in places

If user was blocked it would say they are active and vise versa

keydmitrij committed:
Remove console log

keydmitrij committed:
Make moderator unable to unblock users

keydmitrij committed:
Hide status filters from moderators

keydmitrij committed:
Remove all user data from store on logout

keydmitrij committed:
Disable unblocking and changing roles button if not admin

keydmitrij committed:
Disable removing users if not admin

keydmitrij committed:
Remove unused object

keydmitrij committed:
Add cell text for actions on users

keydmitrij committed:
Add tooltip on user action buttons

keydmitrij committed:
Increase action column width so dropdown menu has enough space

keydmitrij committed:
Move util function outside of jsx

keydmitrij committed:
Remove unused import

keydmitrij committed:
Make only search debounced

keydmitrij committed:
Remove repeating value for  padding

keydmitrij committed:
Remove redundant object

keydmitrij committed:
Move validation fn outside of component

keydmitrij committed:
Remove props that do nothing

keydmitrij committed:
Move getting role tag color fn out of columns type declaration

keydmitrij committed:
Fix a bug with wrong user roles selection by default

keydmitrij committed:
Add destroy modal on hidden

Prevents storing unused and hidden modal windows in memory

keydmitrij committed:
Remove unused const

keydmitrij committed:
Fix a bug with wrong roles being applied

keydmitrij committed:
Add name for user search input

keydmitrij committed:
Fix sign up bug

It would lead to stuck app if phone number is not specified

seledkov commented on src/App.tsx:
```diff
@@ -45,7 +47,15 @@ const router = createBrowserRouter([
+          },
+          {
+            path: "users/:userId",
+            element: <ProfilePage />,
```
UserPage

seledkov commented on src/App.tsx:
```diff
@@ -45,7 +47,15 @@ const router = createBrowserRouter([
           },
           { path: "tasks", element: <TodoListPage /> },
-          { path: "profile", element: <ProfilePage /> },
+          { path: "profile", element: <CurrentProfilePage /> },
```
ProfilePage

seledkov commented on src/api/admin.ts:
```diff
@@ -0,0 +1,135 @@
+      headers: {
+        accept: "application/json",
+      },
+      params: { ...queryParams },
```
params: queryParams
нет особого смысла делать неглубокую копию параметра функции

seledkov commented on src/utility/date.ts:
```diff
@@ -0,0 +1,4 @@
+export const formatDateFromIsoString = (isoDateString: string) => {
```
заюзай библиотеку date-fns

seledkov commented on src/utility/validation.ts:
```diff
@@ -35,6 +35,11 @@ export const emailValidationRules = {
   message: "Must be valid phone number!",
 };
+
+export const hasValidOrder = (str: string | undefined) => {
```
str - типизация некоректная сделай юнион вместо стринги

seledkov commented on src/store/usersTable.ts:
```diff
@@ -0,0 +1,26 @@
+import { createSlice, type PayloadAction } from "@reduxjs/toolkit";
+import type { UserFilters } from "../types/admin";
+
```
тейбл это реализация отображения а у тебя хранятся данные, просто users

seledkov commented on src/pages/app/ProfilePage.tsx:
```diff
@@ -1,90 +1,278 @@
-import { useDispatch } from "react-redux";
-import { authActions } from "../../store/auth";
-import authService from "../../services/authService";
+import { diff } from "deep-object-diff";
```
нельзя просто так затаскивать либы, это надо согласовывать напиши свою функцию собественную которая будет принимать оригинал и чтото измененое и возвращать объект только измененых полей

используй дженерик и утилит тайпы для типизации функции и того что она возращает

seledkov commented on src/pages/app/ProfilePage.tsx:
```diff
@@ -1,90 +1,278 @@
+          (diff(prevUserData, formData) as typeof prevUserData) ||
+          typeof formData;
+
+        if (isEmpty(updatedUserData)) {
```
не тащим либы без согласования

seledkov commented on package.json:
```diff
@@ -13,10 +13,14 @@
     "email-validator": "^2.0.4",
     "libphonenumber-js": "^1.12.9",
+    "lodash": "^4.17.21",
+    "lodash.debounce": "^4.0.8",
```
напиши свою реализацию функцию а не тащи либу без согласования

seledkov replied:
на собесах часто просят написать самому дебаунс

keydmitrij committed:
Change name of user profile page

keydmitrij committed:
Change name for current user profile page

keydmitrij committed:
Remove unused object spreading

keydmitrij committed:
Remove self made date formatting function and add and use date-fns

keydmitrij committed:
Change formatting argument in users page

keydmitrij committed:
Rewrite validation function to have correct typing

keydmitrij committed:
Remove lodash and create isObjectEmpty function

keydmitrij committed:
Create object diff function and use it instead of deep diff

keydmitrij committed:
Rename usersFilter store to users

keydmitrij committed:
Remove lodash debounce and use my own debounce fn

keydmitrij committed:
Add missing export

keydmitrij committed:
Rewrite regular fn to arrow fn

keydmitrij committed:
Change fn name to fit codestyle

keydmitrij committed:
Disable save button on user page if currently is saving

kybikchidori1 commented on src/layouts/AppLayout.tsx:
```diff
@@ -29,38 +34,67 @@ const headerStyle: CSSProperties = {
   const navigate = useNavigate();
   const location = useLocation();
   const isRootURL = location.pathname.match(/\/app\/?$/g);
+  const handleCollapse = () => {
```
Может от  const [headerText, setHeaderText] = useState<string>("To-Do List!") избавиться? А функция handleCollapse будет выглядеть вот так const handleCollapse = () => setIsCollapsed(!isCollapsed);

В вёрстке  {isCollapsed ? "To Do List!" : "To-Do List!"} напрямую, или тебе для чего-то другого нужен этот стейт?

keydmitrij committed:
Remove unnecessary logic for rendering app title and change variable names

keydmitrij committed:
Remove offset in api request when current page is 1

It used to render no info if response data array was shorter than pagination
size
